src/analysis/normed_space/inner_product.lean

```diff
@@ -1734,6 +1773,35 @@ begin
   simp
 end
 
+/-- If `K` is complete, `K` and `K.orthogonal` span the whole space. Version using `complete_space`.
+-/
+lemma submodule.sup_orthogonal_of_complete_space {K : submodule 𝕜 E} [complete_space K] :
+  K ⊔ K.orthogonal = ⊤ :=
+submodule.sup_orthogonal_of_is_complete (complete_space_coe_iff_is_complete.mp ‹_›)
+
+/-- If `K` is complete, any `v` in `E` can be expressed as a sum of elements of `K` and
+`K.orthogonal`. -/
+lemma submodule.exists_sum_mem_mem_orthogonal (K : submodule 𝕜 E) [complete_space K] (v : E) :
+  ∃ (y ∈ K) (z ∈ K.orthogonal), y + z = v :=
```

Comment:
I would write it as `v = y + z` instead of `y + z = v`. Most of the time, you will use this lemma by rcasing, and then you will want to rewrite `v` as `y + z`, so let's have the lemma give the equality in the right direction to do this.

Reply:
Good idea, thanks!